pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -62,13 +62,20 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context, oc *api
 
 	// TODO: pre-check that the cluster domain doesn't already exist
 
+	// fetch vnet and pass it to the subsequent networking checks
 	spAuthorizer, err := dv.validateServicePrincipalProfile(ctx, oc)
 	if err != nil {
 		return err
 	}
+	spVnet := network.NewVirtualNetworksClient(r.SubscriptionID, spAuthorizer)
+
+	vnet, err := spVnet.Get(ctx, r.ResourceGroup, r.ResourceName, "")
```

Comment:
As per a suggestion in our post scrum discussion, I pulled the vnet out and into the main dynamic function in order to be able to pass it to the permission functions.  This made things slightly easier.

Reply:
+1.  I am worried about whether this is going to give us erroneous Forbidden errors though.  cc @nilsanderselde and #632

Reply:
I added a retry here https://github.com/Azure/ARO-RP/pull/632/files#diff-9190e07a0261911419b3f0b2dc20e2f3R82

But now I'm thinking it should be caught at a higher level

Reply:
I think it might work to have the retry in `Create`: https://github.com/Azure/ARO-RP/pull/632/files#diff-b6154a6c303d86ea89ce1c187bc7e01cR48